askama_shared/src/parser.rs

```diff
@@ -312,10 +312,12 @@ fn expr_var_call(i: &[u8]) -> IResult<&[u8], Expr> {
 }
 
 fn path(i: &[u8]) -> IResult<&[u8], Vec<&str>> {
+    let root = opt(value("::", ws(tag("::"))));
```

Comment:
How about we use an empty string instead of "::"? That way, the code generator can remain as it is (a fair bit simpler).

Reply:
Damn... Brain fart! 😄